.github/workflows/cli.lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Code Quality Check
```

Comment:
If we want to use GHA for pre-commit rather than the pre-commit CI (https://pre-commit.ci/) than we should remove the CI configuration from the pre-commit config. I personally would prefer the pre-commit CI because it's really useful, and it's the direction I want to take benchmark-wrapper, but up to y'all.

Reply:
Does pre-commit CI require each dev to install/setup pre-commit on their machines? TBH I want both. I want the pre-commit hooks so that if I enable it I can catch my lint errors earlier, but I want to ensure that every PR gets linted anyways

Reply:
Yep, yeah the only thing that the pre-commit CI will do is replace the Github Workflow. All devs should still install, setup and use pre-commit. The pre-commit CI will run on each push and on each sync of a pull-request, auto-fix pull-requests, create PRs that auto-fix code when possible, and create PRs to upgrade hook versions when possible. Additionally it's really fast because they do a lot of install caching. 

And to clarify, by "auto-fix" I mean that if a hook modifies files during its execution, pre-commit will take those modifications and submit a PR for them.

Reply:
Could we use the pre-commit CI to fix the PR before merging? I really just want PRs to not get merged unless they've been linted. If we can do that then i'm fine with not using GHA